labs/SAW/Game/DLC/proof/Game.py

```diff
@@ -60,78 +60,72 @@ def specification (self):
 
 
 # initDefaultPlayer Contract
-# uint32_t initDefaultPlayer(player_t* player)
+# player_t* initDefaultPlayer()
 class initDefaultPlayer_Contract(Contract):
   def specification (self):
-    # Pull the struct from the bitcode
-    # Although the function uses player_t, pass character_t to SAW since
-    # player_t is an alternative typedef name for character_t.
-    player = self.alloc(alias_ty("struct.character_t"))
+    # No input arguments, so no need to define variables now...
 
     # Symbolically execute the function
-    self.execute_func(player)
+    self.execute_func()
 
-    # Assert the postcondition behaviors
+    # Allocate memory for the player_t struct
+    # Note: Although the function uses player_t, pass character_t to SAW since
+    #       player_t is an alternative typedef name for character_t.
+    player = self.alloc(alias_ty("struct.character_t"))
```

Comment:
Is there a way to figure that out? for example, perhaps we could demonstrate printing the llvm module from within saw, and showing the available structs. It may also be worthwhile to point out that this is not a `saw` problem -- `player_t` isn't available because the C -> LLVM compilation process doesn't maintain that information). Really nail down the fact that SAW evaluates LLVM bytecode.

```
sawscript> r <- llvm_load_module "artifacts/Game.bc"
sawscript> print r
[19:17:09.649] Module: artifacts/Game.bc
Types:
  %struct.character_t = type { [12 x i8], i32, i32, i32, i32, i32 }
  %struct.sprite_t = type { %struct.character_t*,
                            [2 x [4 x [3 x i8]]], i32, i32 }

Globals:

External references:
  declare default void @llvm.dbg.declare(metadata, metadata,
                                         metadata)

Definitions:
  i32 @initDefaultPlayer(%struct.character_t* %0)
  i32 @initDefaultSprite(%struct.character_t* %0,
                         %struct.sprite_t* %1)
  void @resolveAttack(%struct.character_t* %0, i32 %1)
  i32 @checkStats(%struct.character_t* %0)


sawscript>
```

Reply:
This discussion would belong in `SAW.md`.

Reply:
That's a really good point @weaversa and is another way to show `llvm_load_module` for debugging.

Discussion added to `SAW.md`!